### ISSUE TITLE
Vue 3 support

### ISSUE THREAD

**Issue by leokolezhuk** [i18next/i18next-parser#617]
## 🚀 Feature Proposal

Currently the usage of vue-template-compiler in this package prevents using the parser in a vue 3 project.

Supporting both vue 2 and vue 3 versions would be very beneficial I believe.
## Example

Please provide an example for how this feature would be used.

**Comment by Toilal** [comment]
Maybe a simple solution would be to make `vue-template-compiler` a peerDependency.

**Comment by leokolezhuk** [comment]
> You are welcome to make a PR to solve the situation! I will review and merge it.

Hi @karellm,

I am open to doing that.
But while thinking about how to approach it best, I am coming to the conclusion that this **VueLexer might not even be needed at all**. Please, correct me if I am wrong on this.

So in Vue SFC files you have the template and script sections that can contain localizable messages. Regardless of where the message is defined, template or script, it will still be wrapped in a function $t() or t(). Therefore, it seems to me that the **JavaScriptLexer alone is enough to deal with this case**. That is how I am using the parser in my project as well - with the same lexer config for js and vue files. There is no need to compile the .vue file as the VueLexer does it, you can just parse it as a normal JS file then. 

The only case that is not covered with this approach is when one would use a i18next vue wrapper like [this one ](https://panter.github.io/vue-i18next/guide/started.html) and uses the directives(HTML attributes) to define localizable messages. But this case is not covered with the current VueLexer either. But if you use the HTML lexer in combination with the JS lexer for a case like this, you should be covered. 

To sum up at this point I actually see no reason for the VueLexer to exist even for vue users. 
Removing it would also remove the nasty vue-template-compiler dependency that prevents usage with vue3. 

What do you think, @karellm ?

**Comment by karellm** [comment]
Should be fixed as of `9.0.0`

**Comment by poicooler** [comment]
I seem to have a slight issue using version 9.0.0 with vue 3.

When using the translation function in a prop, it is not detected by the parser.

Exemple in some vue files:
```
<template>
  <button :aria-label="t('button aria label')">
    {{ t('button label') }}
  </button>
</template>

...
```

In this case, "button label" is detected and thats fine. However, "button aria label" is not detected and is ignored.

**Comment by daniel100097** [comment]
Does anyone have a solution for parsing attribute translations ? 

like this:

:attr="t('languagekey')"

**Comment by PBK-B** [comment]
> I seem to have a slight issue using version 9.0.0 with vue 3.我在 vue 9.0.0 中使用 3 版似乎遇到了一个小问题。
> 
> When using the translation function in a prop, it is not detected by the parser.在 prop 中使用 translation 函数时，解析器不会检测到它。
> 
> Exemple in some vue files:以一些 vue 文件为例：
> 
> ```
> <template>
>   <button :aria-label="t('button aria label')">
>     {{ t('button label') }}
>   </button>
> </template>
> 
> ...
> ```
> 
> In this case, "button label" is detected and thats fine. However, "button aria label" is not detected and is ignored.在这种情况下，检测到 “button label” ，这很好。但是，“按钮 aria 标签”未被检测到，将被忽略。

@karellm  According to your description, it feels like this matching should be effective, but I still can't get it correctly when I test this usage. Is there any progress on this?

**Comment by geertklop** [comment]
@PBK-B Thanks for your PR and I have locally tested #1094 in our projects and it seems to fix the issues mentioned above. For now I have adjusted that custom lexer for our own codebase.

@karellm any plans or update on merging PR #1094 or any other fixes.
